context/app/static/js/components/workspaces/ConfirmDeleteWorkspacesDialog/ConfirmDeleteWorkspacesDialog.tsx

```diff
@@ -1,96 +1,49 @@
-import DialogTitle from '@mui/material/DialogTitle';
-import Divider from '@mui/material/Divider';
-import IconButton from '@mui/material/IconButton';
-import CloseRounded from '@mui/icons-material/CloseRounded';
+import React from 'react';
+import { useEventCallback } from '@mui/material/utils';
 
 import { SelectedItems } from 'js/hooks/useSelectItems';
-import { generateCommaList } from 'js/helpers/functions';
-import { MergedWorkspace } from 'js/components/workspaces/types';
 import { useWorkspaceToasts } from 'js/components/workspaces/toastHooks';
+import { useWorkspacesList } from 'js/components/workspaces/hooks';
+import ConfirmationDialog from 'js/shared-styles/dialogs/ConfirmationDialog';
+import { getSelectedWorkspaceNames } from 'js/components/workspaces/utils';
 
 interface ConfirmDeleteWorkspacesDialogProps {
-  dialogIsOpen: boolean;
   handleClose: () => void;
-  handleDeleteWorkspace: (workspaceId: number) => Promise<void>;
   selectedWorkspaceIds: SelectedItems;
-  workspacesList: MergedWorkspace[];
 }
 export default function ConfirmDeleteWorkspacesDialog({
-  dialogIsOpen,
   handleClose,
-  handleDeleteWorkspace,
   selectedWorkspaceIds,
-  workspacesList,
 }: ConfirmDeleteWorkspacesDialogProps) {
   const { toastErrorDeleteWorkspaces, toastSuccessDeleteWorkspaces } = useWorkspaceToasts();
+  const { workspacesList, handleDeleteWorkspace } = useWorkspacesList();
 
-  const selectedWorkspaceNames = Array.from(selectedWorkspaceIds).map((id) => {
-    const workspace = workspacesList.find((w) => w.id === Number(id));
-    return workspace ? workspace.name : '';
-  });
-
-  const selectedWorkspaceNamesList = generateCommaList(selectedWorkspaceNames);
+  const selectedWorkspaceNames = getSelectedWorkspaceNames({ selectedWorkspaceIds, workspacesList });
 
-  const handleDeleteAndClose = useCallback(() => {
+  const handleDeleteAndClose = useEventCallback(() => {
     const workspaceIds = [...selectedWorkspaceIds];
 
     Promise.all(workspaceIds.map((workspaceId) => handleDeleteWorkspace(Number(workspaceId))))
       .then(() => {
-        toastSuccessDeleteWorkspaces(selectedWorkspaceNamesList);
+        toastSuccessDeleteWorkspaces(selectedWorkspaceNames);
         selectedWorkspaceIds.clear();
       })
       .catch((e) => {
-        toastErrorDeleteWorkspaces(selectedWorkspaceNamesList);
+        toastErrorDeleteWorkspaces(selectedWorkspaceNames);
         console.error(e);
       });
 
     handleClose();
-  }, [
-    handleDeleteWorkspace,
-    selectedWorkspaceIds,
-    toastSuccessDeleteWorkspaces,
-    toastErrorDeleteWorkspaces,
-    selectedWorkspaceNamesList,
-    handleClose,
-  ]);
+  });
 
   return (
-    <Dialog
-      open={dialogIsOpen}
-      onClose={handleClose}
-      scroll="paper"
-      aria-labelledby="delete-workspace-dialog"
-      maxWidth="lg"
+    <ConfirmationDialog
+      title={`Delete Workspace ${selectedWorkspaceIds.size > 1 ? 's' : ''}`}
```

Comment:
Tiny fix - the plural case would be displayed as `Delete Workspace s` due to the extra space.
```suggestion
      title={`Delete Workspace${selectedWorkspaceIds.size > 1 ? 's' : ''}`}
```